### PR TITLE
chore: added crd configuration reconciliation e2e test

### LINKED ISSUE #236: Create e2e test verifying CR reconciliation
Write a test verifying the flow of changing a flag value in a CRD, applying it and asserting that the flag update is exposed by flagd.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

Applies changes to CRD and asserts that the changes are reflected in a curl to the nginx reverse proxy to flagd.

### Related Issues

Fixes #236

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: ghcr.io/open-feature/open-feature-operator
```
Comment (beeme1mr) [inline config/manager/kustomization.yaml:15]: Is this a regression? I thought we needed `newName` to be controller.

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ spec:
         name: manager
         env:
         - name: FLAGD_VERSION
-          value: INPUT_FLAGD_VERSION
+          value: v0.2.5
```
Comment (beeme1mr) [inline config/manager/manager.yaml:39]: Why was this changed made?

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (skyerus) [reply config/manager/kustomization.yaml]: Accidental, removed

File: config/manager/manager.yaml (hunk 2 above)
Comment (skyerus) [reply config/manager/manager.yaml]: Accidental, removed

**Review by beeme1mr — APPROVED** [review]

**Review by james-milligan — APPROVED** [review]
